### PR TITLE
Implementation some problem from MGH and Hock-Schittkowski

### PR DESCRIPTION
**Pull request by Cebola94**

### CODE REVIEW

File: src/beale.jl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#   See also Buckley#89.
+#   SIF input: Ph. Toint, Dec 1989.
+#
+#   classification SUR2-AN-2-0
+#
+#   A. Cebola, Paraná 10/2016.
+
+export beale
+
+'Beale Model'
```
Comment (dpo) [inline src/beale.jl]: We use double quotes everywhere, except for single characters.

File: src/beale.jl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+function beale(args...)
+
+    nlp = Model()
+    x0 = [1.0, 1.0]
+
+
```
Comment (dpo) [inline src/beale.jl]: There is an extra blank line here and in several other files.

File: src/beale.jl [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    @variable(nlp, x[i=1:2], start=x0[i])
+
+    @NLobjective(
+      nlp,
+      Min,
+      (-1.5+x[1]*(1.0-x[2]))^2 + (-2.25+x[1]*(1.0-x[2]^2))^2 + (-2.625+x[1]*(1.0-x[2]^3))^2
```
Comment (dpo) [inline src/beale.jl]: Please be consistent with the spacing; I prefer a space around all arithmetic operators (+, -, *, /), except to denote negative numbers.

File: src/bt1.jl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+export bt01
+
+'BT01 Model'
+
+function bt01(args...)
+
+    nlp = Model()
+    x0 = [0.08, 0.06]
+
+    M = 13
```
Comment (dpo) [inline src/bt1.jl]: `M` isn't used.

File: src/bt1.jl [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    @NLobjective(
+      nlp,
+      Min,
+      100*x[1]^2+100*x[2]^2-x[1]-100
+    )
+
+    @NLconstraint(
+      nlp,
+      constr1,
+      x[1]^2+x[2]^2-1.0 ==0
```
Comment (dpo) [inline src/bt1.jl]: Spase missing after `==`.

File: src/hs9.jl [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+export hs9
+
+'HS9 Model'
+
+function hs9(args...)
+
+    nlp = Model()
+    x0 = [2.0, 1.0]
+
+    pi = 3.14159;
```
Comment (dpo) [inline src/hs9.jl]: In Julia, you can use a full-precision value of pi.

File: src/zangwil3.jl [hunk 7]
```diff
@@ -0,0 +1,49 @@
+#   Source: problem 1 in
+#   P.T. Boggs and J.W. Tolle,
+#   "A strategy for global convergence in a sequential
+#    quadratic programming algorithm",
+#   SINUM 26(3), pp. 600-623, 1989.
+#
+#   Problem from CUTE: https://github.com/mpf/Optimization-Test-Problems/blob/master/cute
```
Comment (dpo) [inline src/zangwil3.jl]: https://github.com/mpf/Optimization-Test-Problems/blob/master/cute/zangwil3.mod

**Comment by dpo** [comment]
Many thanks for your contribution!

**Comment by Cebola94** [comment]
Sorry Dominique, i'll fix these errors.

File: src/hs27.jl [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# Hock and Schittkowski problem number 26.
+#
```
Comment (dpo) [inline src/hs27.jl]: 27?

File: src/hs27.jl [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+#   Source:
+#   Problem 26 in
+#   W. Hock and K. Schittkowski,
```
Comment (dpo) [inline src/hs27.jl]: 27

File: src/beale.jl [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+function beale(args...)
+
+    nlp = Model()
+
+    @variable(nlp, x[i=1:2], start=1.0)
+
+    @NLobjective(
+      nlp,
+      Min,
+      (-1.5 + x[1] * (1.0 - x[2]))^2 + (-2.25 + x[1] * (1.0 - x[2]^2))^2 + (-2.625 + x[1] * (1.0 - x[2]^3))^2
```
Comment (dpo) [inline src/beale.jl]: Just curious to know why you decided to change the sign of the residuals, compared to the paper?!

File: src/hs9.jl [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+# A. Cebola, Curitiba , 10/2016.
+
+export hs9
+
+"HS9 Model"
+function hs9(args...)
+
+    nlp = Model()
+
+    x0 = [2.0, 1.0]
```
Comment (dpo) [inline src/hs9.jl]: In the HS book, the initial point is (0, 0).

**Comment by Cebola94** [comment]
Errors fixed

**Comment by dpo** [comment]
Thank you!

**Comment by Cebola94** [comment]
Thanks for the opportunity.

File: src/hs27.jl [hunk 12]
```diff
@@ -0,0 +1,36 @@
+# Hock and Schittkowski problem number 27.
+#
+#   Source:
+#   Problem 27 in
+#   W. Hock and K. Schittkowski,
+#   Test examples for nonlinear programming codes,
+#   Lectures Notes in Economics and Mathematical Systems 187,
+#   Springer Verlag, Heidelberg, 1981.
+#
+#   classification QR2-AN-3-1
```
Comment (dpo) [inline src/hs27.jl]: This should be `SQR2-AN-3-1`. Would you mind correcting it in a new pull request? Sorry I didn't notice it earlier. Thanks!

**Pull request merged**